src/main/kotlin/gg/flyte/twilight/time/Time.kt

```diff
@@ -3,18 +3,16 @@ package gg.flyte.twilight.time
 /**
  * Returns the English suffix for a given day of the month.
  *
- * @param day The day of the month for which to retrieve the suffix. Should be a value between 1 and 31.
- * @return The English suffix for the given day of the month.
- * @throws ArrayIndexOutOfBoundsException if the provided day is not within the valid range (1 to 31).
+ * @param day The day of the month for which to retrieve the suffix.
+ * @return The English suffix for the given day.
  */
 fun getDaySuffix(day: Int): String {
```

Comment:
Not sure I rate changing this from an array? I think the array should have been pulled out and made a constant value instead. 

If you want to improve it to make it more efficient, a map would get us down to a couple hundred nanoseconds here rather than the current several thousand-ten thousand we're currently at lmao:

![image](https://github.com/flytegg/twilight/assets/43449531/570b0d20-4906-4c44-ba07-1d34961b5fd5)

(newNew being with a map)

Reply:
Ah, i see i made it so that it wouldn't be capped at 31, the when statement makes it able to worth in the thousands so you could for example 305th

Reply:
That makes sense, didn't think about that as a reason for changing it haha. Maybe that should be another method and then we can have a lightweight one which just supports days of the month? This was what I was using for the loose benchmarks above:

```kt
const val DEFAULT_SUFFIX = "th"

val suffixMap = mapOf(
    1 to "st",
    2 to "nd",
    3 to "rd",
    21 to "st",
    22 to "nd",
    23 to "rd",
    31 to "st"
)

fun newNewGetDaySuffix(day: Int): String = suffixMap[day] ?: DEFAULT_SUFFIX
```

Reply:
it looks quite strange but this is about 5x faster on average
```kotlin
fun getDaySuffix(day: Int): String {
    if (day in 11..13) return "th"

    val lastTwoDigits = day % 100
    val suffix = when (lastTwoDigits) {
        11, 12, 13 -> "th"
        else -> {
            val lastDigit = day % 10
            when (lastDigit) {
                1 -> "st"
                2 -> "nd"
                3 -> "rd"
                else -> "th"
            }
        }
    }

    return suffix
}
```

Reply:
Yes I love micro-optimising lmaoo, sure let's go with that :)

Reply:
Alright committed the change